### PR TITLE
Add the notion of read-only shared volumes

### PR DESCRIPTION
**Pull request by doanac**
This can be really useful for CI jobs that can benefit from something
like a shared public cache.

Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
@kostyapenkov @mike-sul - this is the first building block to start allowing our CI builds to have access to some read-only shared cache. gavelci is running this code so that I could have the PR actually test itself.

@kostyapenkov - this implies we'll eventually be changing the jobserv worker's settings.conf file. I'm not sure if this is "anisble"-able or not? Basically each worker would need something like 
~~~
[shared-volumes]
lmp-cache: ....
~~~
appended to it.

File: examples/projects/shared-volume.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# An example to test a Python based project triggered by GitHub Pull Requests
+# This can be tested under the simulator with:
+#   ./simulator.py create \
+#       -d ./examples/projects/persistent-volume.yml \
```
Comment (kostyapenkov) [inline examples/projects/shared-volume.yml:4]: `persistent-volume.yml`
looks like a leftover from copy-paste

File: examples/projects/shared-volume.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+scripts:
+  flake8: |
+    #!/bin/sh -ex
+    pip3 install flake8
+    flake8 --exclude=migrations/ ./
+
+  unit-test: |
+    #!/bin/sh -ex
+    echo "This is persistent" >> /var/cache-venv/tmp.log
```
Comment (kostyapenkov) [inline examples/projects/shared-volume.yml:43]: "This is shared" instead of  "This is persistent", right?

Comment (kostyapenkov) [reply examples/projects/shared-volume.yml:43]: If runner mount volume in RO-mode, this echo will fail. Am I right?

File: jobserv/project-schema.yml [hunk 3]
```diff
@@ -126,6 +126,12 @@ mapping:
                     mapping:
                       regex;(^[a-zA-Z0-9_\.\-]+$):
                         type: str
+                  shared-volumes:
```
Comment (kostyapenkov) [inline jobserv/project-schema.yml:129]: <nerd_mode>
what about only one type of volumes - just persistent-volumes, but with additional 'mode' attribute: every volume has name, mount-point and mode(RW by default) and RO mode can be used shared volumes.

This approach requires more code-changes, but it is just an idea.
</nerd_mode>

**Comment by kostyapenkov** [comment]
> @kostyapenkov - this implies we'll eventually be changing the jobserv worker's settings.conf file. I'm not sure if this is "anisble"-able or not? Basically each worker would need something like
> 
> ```
> [shared-volumes]
> lmp-cache: ....
> ```
> 
> appended to it.

Until just workers w/o LXD have shared volumes - its easy to manage by ansible, no worries.

File: examples/projects/shared-volume.yml (hunk 1 above)
Comment (doanac) [reply examples/projects/shared-volume.yml]: force-push fixed

File: examples/projects/shared-volume.yml (hunk 2 above)
Comment (doanac) [reply examples/projects/shared-volume.yml]: force-push fixed

File: jobserv/project-schema.yml (hunk 3 above)
Comment (doanac) [reply jobserv/project-schema.yml:129]: persistent read-write volumes work a little differently. They don't have to configured by the host up front and are instead nested under the project in a dynamic self-contained way.

**Pull request merged**